src/Pages/CoinDetails.js

```diff
@@ -0,0 +1,57 @@
+import { useLocation, Navigate } from 'react-router-dom';
+
+const CoinDetails = () => {
+  const location = useLocation();
+
+  const singleData = location?.state?.coin;
+  if (!singleData) {
+    return <Navigate to="/" replace />;
+  }
+  return (
+    <div>
+      <div className="container-detail">
+        <p className="coin-supply">Coins supply in market:</p>
+        <p className="coin-supply">
+          $
+          {singleData.csupply}
+        </p>
+      </div>
+      <ul>
+        <li className="name">
+          <span>name :</span>
+          {' '}
+          {singleData.name}
+        </li>
+        <li className="name">
+          <span>Rank :</span>
+          {' '}
+          {singleData.rank}
+        </li>
+        <li className="name">
+          <span>Symbol :</span>
+          {' '}
+          {singleData.symbol}
+        </li>
+        <li className="name">
+          <span>Market cap in USD :</span>
+          {' '}
+          $
+          {singleData.market_cap_usd}
+        </li>
+        <li className="name">
+          <span>Weekly percentage change :</span>
+          {' '}
+          {singleData.percent_change_7d}
+          %
+        </li>
+        <li className="name">
+          <span>Price :</span>
+          {' '}
+          $
+          {singleData.price_usd}
+        </li>
+      </ul>
+    </div>
+  );
+};
```

Comment:
- [ ] Kindly make sure you update the Details page by using different **background colors** on the `even` and `odd` rows detail element respectively to match the given template design. Below I used an image example of the pink color but you should use your chosen color.

| Template Detail Page | Your Detail Page design| 
| ----------------------------- | ---------------------------------- |
| ![Screenshot from 2022-09-09 19-26-13](https://user-images.githubusercontent.com/83042742/189465379-582ea04f-f40e-4c42-9b5f-6ec0d5d60fc7.png) | ![Screenshot from 2022-09-10 04-26-19](https://user-images.githubusercontent.com/83042742/189465369-570af5da-f2c6-4ecd-99e3-d6b17da6a274.png) |